src/sentry/options/manager.py

```diff
@@ -81,7 +81,15 @@ def set(self, key, value):
         assert not (opt.flags & FLAG_PRIORITIZE_DISK and settings.SENTRY_OPTIONS.get(key)), '%r cannot be changed at runtime because it is configured on disk' % key
 
         if not isinstance(value, opt.type):
```

Comment:
You should use your `_value_is_of_type` here, no?

If we set it to basestring, you're saying the type becomes `unicode`, so what happens if it's being set as `str`? I don't think we'd want to lock into unicode vs str explicitly imo.